File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let intSet = new Set();
        let isDuplicate = false;
        nums.forEach((num) => {
            if (!intSet.has(num)) {
                intSet.add(num);
            } else {
                isDuplicate = true;
            }
        })
        return isDuplicate;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] output = new int[nums.length];
        for (int i = nums.length - 1; i >= 0; i--) {
            int length = nums.length - 1;
            output[i] = 1;
            while (length >= 0) {
                if (i != length) {
                    output[i] = output[i] * nums[length];
                }
                length--;
            }
        }

        return output;
    }
}  
